mozilla/network-pulse-api
# Move bookmarks from EmailUser to UserProfile

Addresses https://github.com/mozilla/network-pulse-api/issues/133

This PR sets up a UserProfile model, and moves user `bookmarks` over from the EmailUser model to this UserProfile model instead. This will let us both start padding out the UserProfile, and clean up the EmailUser to the point where we can hopefully replace it with a Django-managed user instead.

How to test this PR:

1. follow the setup instructions in the README for making sure you can use/log in
2. grab this database file: [db.tar.gz](https://github.com/mozilla/network-pulse-api/files/1201641/db.tar.gz) and unpack it to the root. This is a database with migrations applied equal to the pre-this-PR codebase, with a bare minimum of faves stored
3. run `manage migrate`, then `manage test`. The migration will likely ask you to delete the stale database users_userbookmarks: that is fine. Running `manage test` should show 41 tests passing.
4. run `manage runserver` and hit up your http://test.example.com:8000/admin and have a look at the profile/bookmark data, as well as what's in the emailuser entries. It should all look kosher.
